### PR TITLE
[GRPO] In-place temperature scaling operation

### PR DESCRIPTION
**Pull request by winglian**
# What does this PR do?

Use in-place operator for logits to save memory. Even for small models like qwen2 0.5B, bsz=8, and completion_len of 256, the logits tensor is ~5GB, and duplicating this for the temperature scaling was using another 5GB un-necessarily.
Screenshot 2026-03-10 at 12 00 28 AM

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

---

> [!NOTE]
> **Low Risk**
> Behavior should be numerically identical but changes tensor mutation semantics; main risk is unexpected aliasing if `logits` were reused elsewhere (it is not in this path).
> 
> **Overview**
> Updates `GRPOTrainer._get_per_token_logps_and_entropies` to apply temperature scaling via in-place `logits.div_(self.temperature)` instead of `logits / self.temperature`, reducing peak memory during logprob/entropy computation for large `(B, T, V)` logits tensors.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b2e8aed7249592acc900bcf50e2f8e1c05a985df. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by qgallouedec** [comment]
nice, thanks!

**Comment by qgallouedec** [comment]
@codex review

**Review by qgallouedec — APPROVED** [review]
